Fixes for KSP 1.5.0

Flupster opened the pull request.

Flupster committed:
Fixes for KSP 1.5.0

Flupster committed:
Revert csproj

MateoCB commented:
This is working nicely, thanks @Flupster

CptBoBy commented:
compiled the mod with the modified files ,still doesn't work.

Flupster commented:
I'll upload the compiled files later tonight and link them here, compiled fine for me and worked

DonCorleone-NL commented:
That would be awesome Flupster, thx alot mate!

new01DAY commented:
I'm not sure, but I think it doesn't work for some because of the changes to [x] Science!.csproj `<HintPath>E:\SteamLibrary\steamapps\common\Kerbal Space Program\KSP_x64_Data\Managed\UnityEngine.dll</HintPath>` from `<HintPath>..\..\..\..\Games\KSP\KSP_Real\KSP_x64_Data\Managed\UnityEngine.dll</HintPath>`.

Flupster commented:
You need to add the references to the DLLs,mine was in another location,however I reverted it back in the second commit

new01DAY commented:
Yeah, sorry. I just realized why/what it was. I actually don't know much about compiling and I'm trying to figure it out.

CptBoBy commented:
compiled without problems having set the proper references , but game still says incompatible and the x science window appears completely empty.

Flupster commented:
zip of the compiled mod can be found in my fork releases over at https://github.com/Flupster/KSP-X-Science/releases/

Flupster commented:
> compiled without problems having set the proper references , but game still says incompatible and the x science window appears completely empty.

Just did a full reinstall of KSP dropped in the mod from my forks release page and works, i'm not sure why yours isn't working but you could try mine?

https://i.imgur.com/qSoSuF9.jpg

CptBoBy commented:
> > compiled without problems having set the proper references , but game still says incompatible and the x science window appears completely empty.
> 
> Just did a full reinstall of KSP dropped in the mod from my forks release page and works, i'm not sure why yours isn't working but you could try mine?
> 
> https://i.imgur.com/qSoSuF9.jpg

tried with my old install , not even the empty science window appeared. completely reinstalled ksp ,and same thing ,at startup game says mod compatible only with 1.4.5 ,and game pretends mod is not even installed

Flupster committed:
KSP 1.5.1 fix

Flupster commented:
> Thanks. But your 5.17.2 [release](https://github.com/Flupster/KSP-X-Science/releases/tag/5.17.2) zip that includes this change is broken - only 22 bytes.

My bad forgot to actually include in the mod within the zip, should be fixed now!

Oeddish commented:
FWIW I came here to submit a PR with more or less your exact changes, which work for me (with minor differences I'll comment on in the diff).  Would suggest also updating CHANGES.md to keep the project consistent.

Oeddish commented on X-Science/GameData/[x] Science!/[x] Science!.version:
```diff
@@ -7,12 +7,12 @@
         "MAJOR":5,
         "MINOR":17,
-        "PATCH":0
+        "PATCH":2
```
I'd suggest this should be a minor revision bump rather than a patch if only because the KSP minor version changed, although I can see an argument either way.

Flupster replied:
No changes were applied to the mod, so realistically it's a patch and not a version change

Edit: seems the previous maintainer did increase the minor revision number on recompilation, to keep with historical changes I've also made the same changes

Oeddish commented on X-Science/ScienceChecklistAddon.cs:
```diff
@@ -835,14 +835,14 @@ public bool Muted
                 {
                     // Set the music volume to what it was before
-                    MusicLogic.SetVolume(oldVolume);
+                    MusicLogic.SetVolume(oldVolume, oldVolume);
```
Not sure if it matters, but i tested this with 0f rather than oldVolume for the parameter.  Seemed to work as expected.

Flupster replied:
I have to be honest, I'm not entirely sure what the second parameter of SetVolume is, it just required a float so I used the first parameter as the second, If you know what the second parameter is let me know and if `0f` is the better option for the second parameter

Flupster committed:
Minor version bump to keep with historical recompiling of newer KSP versions

Tekaoh2 commented:
This works on my machine. +1 vote for merge so it can go on CKAN

Flupster committed:
AssemblyInfo update forgot to changefor v5.18

Flupster committed:
Change versioning for KSP 1.6.0

Flupster commented:
Updated for v1.6.0 of KSP, no actual changes were made just the version numbers, tested a little all seems to be working.

SilkDev commented:
Hey @Flupster, it seems there's no maintainers left here, maybe you could push it to CKAN as a new package ?

Or maybe a little ping to @thewebbooth could work ?

Flupster committed:
Changes for v1.6.1 of KSP also update .version for CKAN

siimav committed:
Fixed performance issues with RP-1

siimav committed:
DMagic API changes

siimav committed:
Add support for the new totalScienceLevel field in DMModuleScienceAnimateGeneric

Flupster committed:
Update [x] Science!.version

This was broken, requests coming from `%255Bx%255D%2520Science%2521/%255Bx%255D%2520Science%2521.version`

siimav committed:
Fixed return statements in IsAmountLimitedByDMagic()

siimav committed:
Fixed RunExperiment() bypassing DMagic module checks

siimav committed:
Fixed CanRunExperiment() for DMModuleScienceAnimateGeneric

Flupster committed:
changes for KSP 1.7.0

Flupster committed:
Forgot to change download URL

siimav committed:
Added a config setting for checking for experiment results on unloaded vessels, is disabled by default

siimav committed:
Updated version file to support KSP versions starting from 1.5

Flupster committed:
Merge pull request #1 from siimav/flupster

Performance improvements & DMagic API fixes

Flupster committed:
v5.22 Release

Flupster committed:
Update CHANGES.md

Flupster committed:
Update CHANGES.md

Flupster committed:
Update [x] Science!.version

Tekaoh committed:
Comment out an outdated assumption

Perhaps these science combinations were not possible at one time, but they are definitely possible now. I've achieved them using the Field Science contract pack to help me with the waypoints. But the contract pack is not required if you know where they are.

Flupster committed:
Merge pull request #2 from Tekaoh/patch-1

Comment out an outdated assumption

IDizor committed:
Here and Now: Created science threshold slider. Added next experiment science value in brackets. Improved performance.

IDizor committed:
Here and Now: Fixed science threshold save/load functionality.

IDizor committed:
Removed fire events debug logging.

IDizor committed:
Fixed performance issues. Implemented acceleration slider.

Commit:
Ignore null exception when gathering experiments

In KSP 1.8.0, some parts throw a NullReferenceException when doing this.
Interestingly, the parts that throw this are the kerbalEVA* parts from
the DLCs when you don't have the DLCs.

Flupster committed:
Merge pull request #6 from trosine/ksp_1.8.0

Ignore null exception when gathering experiments

Flupster committed:
Changes for 1.8

Flupster committed:
v5.24 recompile for 1.9 ksp

Flupster committed:
My bad, typo in min version

Flupster committed:
Merge pull request #5 from IDizor/master

Science threshold feature for Here and Now window.

Flupster committed:
Version bump
